src/connection.c

```diff
@@ -566,14 +610,51 @@ int keep_trying_to_connect (create_connection_ctx_t *ctx,
 
       if (rtn == CONN_WAIT_ACTION_RETRY) // if redirected or build_headers
         continue;
-      backoff_delay (backoff_timer); // 3,7,15,31 ..
+      // If interface down event is set, stop retry
+      // and wait till interface is up again.
+      if(get_interface_down_event()) {
+	    if (0 != wait_while_interface_down())
+	      return false;
+	    start_conn_in_progress();
+	    ParodusInfo("Interface is back up, re-initializing the convey header\n");
+	    // Reset the reconnect reason by initializing the convey header again
+	    ((header_info_t *)(&ctx->header_info))->conveyHeader = getWebpaConveyHeader();
+	    ParodusInfo("Received reconnect_reason as:%s\n", reconnect_reason);  
+      } else { 
+        if (backoff_delay (backoff_timer) // 3,7,15,31 ..
+              != BACKOFF_DELAY_TAKEN) // shutdown or cond wait error
+          return false;
+      }
       if (rtn == CONN_WAIT_RETRY_DNS)
         return false;  //find_server again
       // else retry
     }
     return false;
 }
 
+int wait_while_interface_down()
+
+{
+	int rtn;
+	
+	ParodusError("Interface is down, hence waiting until its up\n");
+	close_and_unref_connection (get_global_conn());
+	set_global_conn(NULL);
+
+	pthread_mutex_lock(get_interface_down_mut());
+	while (!get_interface_down_event ()) {
+  	  rtn = pthread_cond_wait(get_interface_down_con(), get_interface_down_mut());
+  	  if (rtn != 0)
+  	    ParodusError 
+  	      ("Error on pthread_cond_wait (%d) in wait_while_interface_down\n", rtn);
+  	  if ((rtn != 0) || g_shutdown) {
+	    pthread_mutex_unlock (get_interface_down_mut());
+	    return -1;
+	  }
+	}
+    pthread_mutex_unlock (get_interface_down_mut());
```

Comment:
I think this unlock should be moved immediately after the 646 pthread_cond_wait line